tests/integration/test_logic.py

```diff
@@ -20,57 +20,125 @@
 
 import cunumeric as num
 
+SCALARS_INF = (np.inf, -np.inf, np.nan, 0)
+ARRAYS_INF = ([np.inf, -np.inf, np.nan, 0],)
 
-def test_inf():
-    arr = [np.inf, -np.inf, np.nan, 0]
-    assert np.array_equal(np.isposinf(arr), num.isposinf(arr))
-    assert np.array_equal(np.isneginf(arr), num.isneginf(arr))
 
+@pytest.mark.parametrize("x", SCALARS_INF + ARRAYS_INF)
+def test_inf_basic(x):
+    assert np.array_equal(np.isposinf(x), num.isposinf(x))
+    assert np.array_equal(np.isneginf(x), num.isneginf(x))
 
-INPUTS = (
+
+@pytest.mark.parametrize("out_dt", (bool, int, float))
+@pytest.mark.parametrize("x", (np.inf,) + ARRAYS_INF)
+@pytest.mark.parametrize("func_name", ("isposinf", "isneginf"))
+def test_inf_out(func_name, x, out_dt):
+    res_shape = (4,)
+    res_np = np.empty(res_shape, dtype=out_dt)
+    res_num = num.empty(res_shape, dtype=out_dt)
+
+    func_np = getattr(np, func_name)
+    func_num = getattr(num, func_name)
+
+    func_np(x, out=res_np)
+    func_num(x, out=res_num)
+    assert np.array_equal(res_np, res_num)
+
+
+class TestInfErrors:
+    @pytest.mark.parametrize("func_name", ("isposinf", "isneginf"))
+    def test_out_invalid_shape(self, func_name):
+        expected_exc = ValueError
+        x = [np.inf, -np.inf, np.nan, 0]
+        res_shape = (3,)
+        res_np = np.empty(res_shape)
+        res_num = num.empty(res_shape)
+
+        func_np = getattr(np, func_name)
+        func_num = getattr(num, func_name)
+
+        with pytest.raises(expected_exc):
+            func_np(x, out=res_np)
+        with pytest.raises(expected_exc):
+            func_num(x, out=res_num)
+
+
+SCALARS = (pytest.param("a string", marks=pytest.mark.xfail), None, False)
+ARRAYS = (
     [1.0, 2.0, 3.0],
     [1.0 + 0j, 2.0 + 0j, 3.0 + 0j],
     [1.0 + 1j, 2.0 + 0j, 3.0 + 1j],
 )
 
 
-@pytest.mark.parametrize("arr", INPUTS)
-def test_predicates(arr):
-    in_np = np.array(arr)
-    in_num = num.array(arr)
+@pytest.mark.parametrize("x", SCALARS + ARRAYS)
+def test_real_complex(x):
+    # for x is 'a string', np.isreal is False, num.isreal is Array(True)
+    assert np.array_equal(np.isreal(x), num.isreal(x))
+    assert np.array_equal(np.iscomplex(x), num.iscomplex(x))
 
-    assert np.array_equal(np.isreal(in_np), num.isreal(in_num))
-    assert np.array_equal(np.iscomplex(in_np), num.iscomplex(in_num))
 
+@pytest.mark.parametrize("x", SCALARS)
+def test_real_complex_object_scalar(x):
+    assert np.array_equal(np.isrealobj(x), num.isrealobj(x))
+    assert np.array_equal(np.iscomplexobj(x), num.iscomplexobj(x))
```

Comment:
suggest split into separate tests for real and complex rather than combining (and test below as well)

Reply:
Fixed. Thanks.